baseplate/lib/message_queue.py

```diff
@@ -111,14 +147,100 @@ def unlink(self) -> None:
         self.queue.unlink()
 
     def close(self) -> None:
-        """Close the queue, freeing related resources.
+        self.queue.close()
 
-        This must be called explicitly if queues are created/destroyed on the
-        fly. It is not automatically called when the object is reclaimed by
-        Python.
 
-        """
-        self.queue.close()
+class InMemoryMessageQueue(MessageQueue):
+    """An in-memory inter process message queue.
+
+    Uses a simple Python Queue to store data.
+
+    Used in conjunction with the RemoteMessageQueue to
+    provide an alternative to Posix queues, for systems
+    that don't have Posix available. The client will
+    instantitate a RemoteMessageQueue, which connect to
+    a Thrift server. The Thrift server internally uses an
+    InMemoryMessageQueue to store data.
+
+    """
+
+    def __init__(self, name: str, max_messages: int):
+        self.queue: q.Queue = q.Queue(max_messages)
+        self.max_messages = max_messages
+        self.name = name
+
+    def get(self, timeout: Optional[float] = None) -> bytes:
+        try:
+            message = self.queue.get(timeout=timeout)
+            self.queue.task_done()
+            return message
+        except q.Empty:
+            raise TimedOutError
+
+    def put(self, message: bytes, timeout: Optional[float] = None) -> None:
+        try:
+            self.queue.put(message, timeout=timeout)
+        except q.Full:
+            raise TimedOutError
+
+    def close(self) -> None:
+        """Not implemented for in-memory queue"""
+
+
+class RemoteMessageQueue(MessageQueue):
+    """A message queue that uses a remote Thrift server.
+
+    Used in conjunction with the InMemoryMessageQueue to
+    provide an alternative to Posix queues, for systems
+    that don't have Posix available.
+
+    """
+
+    def __init__(self, name: str, max_messages: int, host: str = "127.0.0.1", port: int = 9090):
+        # Connect to the remote queue server, and creeate the new queue
+        self.name = name
+        self.max_messages = max_messages
+        self.host = host
+        self.port = port
+        self.connect()
+        self.client.create_queue(name, max_messages)
+
+    def connect(self) -> None:
+        # Establish a connection with the queue server
+        transport = TSocket.TSocket(self.host, self.port)
+        self.transport = TTransport.TBufferedTransport(transport)
+        protocol = TBinaryProtocol.TBinaryProtocol(self.transport)
+        self.client = RemoteMessageQueueService.Client(protocol)
+        self.transport.open()
+
+    def get(self, timeout: Optional[float] = None) -> bytes:
+        # Call the remote server and get an element for the correct queue
+        try:
+            try:
```

Comment:
Instead of a nested try you can just have multiple except statements below the first try. That would be easier to read.

Reply:
Sorry, I see that you are trying to catch a possible timeout error after trying to reconnect. In that case I would wrap the inner try in a private (eg `__try_to_get(self)`) and put that inside the outer try.